src/formElementPlugins/radioChoice/src/ApplicationView.tsx

```diff
@@ -1,59 +1,123 @@
     // This ensures that, if a default is specified, it gets saved on first load
-    if (!value && defaultOption !== undefined) {
-      const defaultIndex = getDefaultIndex(defaultOption, options)
-      const defaultValue = options?.[defaultIndex]
+    if (!currentResponse?.text && defaultOption !== undefined) {
+      const optionIndex = getDefaultIndex(defaultOption, options)
       onSave({
-        text: defaultValue,
-        optionIndex: defaultIndex,
+        text: optionsDisplayProperty
+          ? allOptions[optionIndex][optionsDisplayProperty]
+          : allOptions[optionIndex],
+        selection: allOptions[optionIndex],
+        optionIndex,
       })
-      setValue(defaultValue)
+    }
+    if (currentResponse?.text) {
+      const { optionIndex } = currentResponse
+      setSelectedIndex(optionIndex)
     }
   }, [])
 
   function handleChange(e: any, data: any) {
-    const { value, index } = data
-    setValue(value)
-    onSave({ text: value, optionIndex: index })
+    const { index: optionIndex } = data
+    setSelectedIndex(optionIndex)
+    onSave({
+      text:
+        hasOther && optionIndex === allOptions.length - 1
+          ? `${strings.OTHER}: ${otherText}`
+          : optionsDisplayProperty
+          ? allOptions[optionIndex][optionsDisplayProperty]
+          : allOptions[optionIndex],
+      selection: allOptions[optionIndex],
+      optionIndex,
+      other: hasOther && optionIndex === allOptions.length - 1 ? otherText : undefined,
+    })
+  }
+
+  function handleOtherChange(e: any, { value }: any) {
+    setOtherText(value)
+  }
+
+  function handleOtherLoseFocus(e: any) {
+    onSave({
+      text: `${strings.OTHER}: ${otherText}`,
+      selection: allOptions[allOptions.length - 1],
+      optionIndex: allOptions.length - 1,
+      other: hasOther ? otherText : undefined,
+    })
   }
 
+  const radioButtonOptions = allOptions.map((option: any, index: number) => {
+    return {
+      key: `${index}_${option}`,
+      text: optionsDisplayProperty ? option[optionsDisplayProperty] : option,
+      value: index,
+    }
+  })
+
   return (
     <>
       <label>
         <Markdown text={label} semanticComponent="noParagraph" />
       </label>
       <Markdown text={description} />
-      {options.map((option: string, index: number) => {
+      {radioButtonOptions.map((option: any, index: number) => {
         return (
           <Form.Field key={`${index}_${option}`} disabled={!isEditable}>
             <Radio
-              label={option}
+              label={option.text}
               name={`${code}_radio_${index}`} // This is GROUP name
-              value={option}
-              checked={option === value}
+              value={selectedIndex}
+              checked={index === selectedIndex}
               onChange={handleChange}
               index={index}
             />
+            {/* TO-DO: Need to find a way to make this input field inline with  the last Other radio button*/}
```

Comment:
<img width="300" alt="Screen Shot 2021-04-06 at 4 12 05 PM" src="https://user-images.githubusercontent.com/16461988/113657831-e1d4be00-96f2-11eb-99e2-ff22a7edd8ce.png">

: ) Just need to do the same check on `<Form.Field (...) inline={hasOther && index === allOptions.length - 1}>` prob better to keep the same check in a variable.

Reply:
Oh that's great -- much nicer. Thanks. :)

Reply:
Done 3a152ee7